search/tasks.py

```diff
@@ -17,3 +17,13 @@ def refresh_index(index_name):
     from search.utils import get_conn
     conn = get_conn()
     conn.indices.refresh(index=index_name)
+
+
+@async.task
+@statsd.timer('lore.search.tasks.index_resources')
+def index_resources(resources):
+    """
+    Reindex the given resources.
+    """
+    from search.utils import index_resources as _index_resources
+    _index_resources(resources)
```

Comment:
All calls to `index_resources` send an iterable of LearningResource instances, which means Celery is serializing and deserializing instances, wasting resources. 

Also, the burden of that query is borne by the user-facing Django-app instead of the Celery task.

In addition, those serialized resources are stale, and the indexing sends them to the `resource_to_dict` function to get the data for Elasticsearch to build the actual index data. That could be out-of-date due to the async delay.

All calls to `index_resources` should send an iterable of integers (primary keys), and `index_resources` should do the query.